cmd/trickster/main.go

```diff
@@ -72,12 +73,47 @@ func main() {
 		log.Fatal(1, err.Error(), log.Pairs{})
 	}
 
-	// Start the Server
-	l, err := proxy.NewListener(config.ProxyServer.ListenAddress, config.ProxyServer.ListenPort,
-		config.ProxyServer.ConnectionsLimit)
+	if config.ProxyServer.TLSListenPort < 1 && config.ProxyServer.ListenPort < 1 {
+		log.Fatal(1, "no http or https listeners configured", log.Pairs{})
+	}
+
+	wg := sync.WaitGroup{}
+
+	// if TLS port is configured and at least one origin is mapped to a good tls config,
+	// then set up the tls server listener instance
+	if config.ProxyServer.ServeTLS && config.ProxyServer.TLSListenPort > 0 {
+		wg.Add(1)
+		go func() {
+			tlsConfig, err := config.Config.TLSCertConfig()
+			if err == nil {
+				l, err := proxy.NewListener(
+					config.ProxyServer.TLSListenAddress,
+					config.ProxyServer.TLSListenPort,
+					config.ProxyServer.ConnectionsLimit,
+					tlsConfig)
+				if err == nil {
+					err = http.Serve(l, handlers.CompressHandler(routing.TLSRouter))
+				}
+			}
+			log.Error("exiting", log.Pairs{"err": err})
+			wg.Done()
+		}()
+	}
```

Comment:
Do we want to log here in the else condition? Want to avoid failing silently.

Reply:
I think we are OK here, in that it should not be possible to fail silently. The condition block starting L66 will fatal out before getting to this point if no listeners are configured, and the configuration loader L48 will fail out earlier than that if TLS is misconfigured; therefore, at this point of the execution, at least one of the listener blocks must execute and they are both logging errors.